pkgs/development/libraries/zlib/default.nix

```diff
@@ -125,6 +126,14 @@ stdenv.mkDerivation rec {
     "SHARED_MODE=1"
   ];
 
+  passthru = {
+    tests = {
+      # NB: this may test a different variant of zlib than the package that
+      #     carries this `tests` attribute, due to overriding.
+      pkg-configPackages = tests.pkg-configPackages.zlib;
```

Comment:
I would rather this call `makePkgConfigTest`, and then `tests.pkg-configPackages` fix these out. That is I think better layering, but concretely it allows the tests for *individual* packages to be built regardless of the package in question.

Reply:
The point is exactly to violate layering, so that ofborg will test the relevant bit of `pkg-configPackages` when it's time to do so. I think this is the best we can achieve with the tooling we have, if we put in the effort.

Reply:
Maybe we should just ignore this and rely on asynchronous testing of the whole `pkg-configPackages` on hydra.

Reply:
Yeah I don't know the details of ofborg, but I thought the overriding was just for the top-level tests? I would think if the individual package is an unsupported systems, then ofborg would just skip the test too, which is fine?

Reply:
I've factored out the tester into a properly tested and documented tester.